src/main/java/oripa/util/gui/GridBagConstraintsBuilder.java

```diff
@@ -0,0 +1,185 @@
+package oripa.util.gui;
+
+import java.awt.GridBagConstraints;
+import java.awt.Insets;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import oripa.view.main.MainFrame;
+
+/**
+ * remove gridBagConstraint management from UI Code by providing simple
+ * functions to return GridBagConstraints of the same layout definitions
+ *
+ * @author lifesbest23
+ *
+ */
+public class GridBagConstraintsBuilder {
+	private int gridX, gridY;
+	private final int gridWidth;
+	private int anchor;
+
+	private double weightX, weightY;
+	private int fill;
+	private Insets insets;
+
+	private static final Logger logger = LoggerFactory.getLogger(MainFrame.class);
```

Comment:
The second suggestion sounds like a good idea, I just copied it from the MainFrame propably still at the beginning of getting to know how everything worked, but that would help debugging UI layout propably.

Reply:
Ok, I eventually landed on somewhat of a middle ground between the two suggestions.
Since most parents are actually JPanels, it would not be of much help to set it only to JPanel, but in case there is a class inheriting from JPanel as a parent, the first constructor can be used.